components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java

```diff
@@ -224,12 +228,77 @@ public String generateAPIDefinition(SwaggerData swaggerData) throws APIManagemen
         info.setVersion(swaggerData.getVersion());
         openAPI.setInfo(info);
         updateSwaggerSecurityDefinition(openAPI, swaggerData);
-        for (SwaggerData.Resource resource : swaggerData.getResources()) {
-            addOrUpdatePathToSwagger(openAPI, resource);
+        if (APIConstants.GRAPHQL_API.equals(swaggerData.getTransportType())) {
+            openAPI = modifyGraphQLSwagger(openAPI);
+        } else {
+            for (SwaggerData.Resource resource : swaggerData.getResources()) {
+                addOrUpdatePathToSwagger(openAPI, resource);
+            }
         }
         return Json.pretty(openAPI);
     }
 
+    /**
+     * Construct openAPI definition for graphQL. Add get and post operations
+     *
+     * @param openAPI OpenAPI
+     */
+    private OpenAPI modifyGraphQLSwagger(OpenAPI openAPI) {
+        SwaggerData.Resource resource = new SwaggerData.Resource();
+        resource.setAuthType("Any");
```

Comment:
You can use the APIConstants.AUTH_APPLICATION_OR_USER_LEVEL_TOKEN here

Reply:
Fixed.